src/go/rpk/pkg/cli/topic/analyze.go

```diff
@@ -0,0 +1,743 @@
+		} else {
+			summary.BatchesPerSecond = append(summary.BatchesPerSecond, percentileValue[float64]{p, 0})
+		}
+
+		if len(batchSizes) != 0 {
+			pBatchSize, err := utils.Percentile(batchSizes, p)
+			if err != nil {
+				return summary, err
+			}
+			summary.BatchSizes = append(summary.BatchSizes, percentileValue[int]{p, pBatchSize})
+		} else {
+			summary.BatchSizes = append(summary.BatchSizes, percentileValue[int]{p, 0})
+		}
+	}
+
+	return summary, nil
+}
+
+type printConfig struct {
+	Percentiles        []float64
+	PrintGlobalSummary bool
+	PrintTopicSummary  bool
+}
+
+func (cr *collectResult) printSummaries(cfg printConfig) error {
+	percentiles := cfg.Percentiles
+	percentileNames := []string{}
+	for _, p := range percentiles {
+		percentileNames = append(percentileNames, fmt.Sprintf("P%v", p))
+	}
+
+	const (
+		secSummary   = "summary"
+		secBatchRate = "partition batch rate (1/s)"
+		secBatchSize = "partition batch size (bytes)"
+	)
+
+	ts, err := cr.summarizeTopics(percentiles)
+	if err != nil {
+		return err
+	}
+	// Print the topic summaries in asc order by topic name
+	slices.SortFunc(ts, func(a topicSummary, b topicSummary) int {
+		return strings.Compare(a.Topic, b.Topic)
+	})
+
+	gs, err := cr.summarizeGlobal([]float64{50})
+	if err != nil {
+		return err
+	}
+
+	sections := out.NewMaybeHeaderSections(
+		out.ConditionalSectionHeaders(map[string]bool{
+			secSummary:   cfg.PrintGlobalSummary,
+			secBatchRate: cfg.PrintTopicSummary,
+			secBatchSize: cfg.PrintTopicSummary,
+		})...,
+	)
+
+	sections.Add(secSummary, func() {
+		tw := out.NewTabWriter()
+		tw.PrintColumn("topics", gs.Topics)
+		tw.PrintColumn("partitions", gs.Partitions)
+		tw.PrintColumn("total batch rate (1/s)", gs.TotalBatchesPerSecond)
+		tw.PrintColumn("p50 batch size (bytes)", gs.BatchSizes[0].value)
+		tw.Flush()
+	})
+
+	sections.Add(secBatchRate, func() {
+		tw := out.NewTable(append([]string{"TOPIC"}, percentileNames...)...)
+		defer tw.Flush()
+		for _, ts := range ts {
+			vals := []interface{}{}
+			for _, pv := range ts.BatchesPerSecond {
+				vals = append(vals, pv.value)
+			}
+
+			tw.Print(append([]interface{}{ts.Topic}, vals...)...)
+		}
+	})
+
+	sections.Add(secBatchSize, func() {
+		tw := out.NewTable(append([]string{"TOPIC"}, percentileNames...)...)
+		defer tw.Flush()
+		for _, ts := range ts {
+			vals := []interface{}{}
+			for _, pv := range ts.BatchSizes {
+				vals = append(vals, pv.value)
+			}
+
+			tw.Print(append([]interface{}{ts.Topic}, vals...)...)
+		}
+	})
+	return nil
+}
+
+type kvTPInfo struct {
+	TopicPartition topicPartition      `json:"topicPartition" yaml:"topic_partition"`
+	Info           *topicPartitionInfo `json:"info" yaml:"info"`
+}
+
+type rawResults struct {
+	TopicPartitionInfo []kvTPInfo `json:"topicPartitionInfo" yaml:"topic_partition_info"`
+	TimeRange          timeRange  `json:"timeRange" yaml:"time_range"`
+}
+
+func (cr collectResult) printRawResults(f config.OutFormatter, w io.Writer) (bool, error) {
```

Comment:
```suggestion
func (cr *collectResult) printRawResults(f config.OutFormatter, w io.Writer) (bool, error) {
```

collectResult has methods on both value and pointer receivers

Reply:
D'oh, nice catch. They should've all been pointer receivers.